src/panels/lovelace/cards/hui-history-graph-card.ts

```diff
@@ -89,38 +86,40 @@ export class HuiHistoryGraphCard extends LitElement implements LovelaceCard {
   public connectedCallback() {
     super.connectedCallback();
     if (this.hasUpdated) {
-      this._subscribeHistoryTimeWindow();
+      this._subscribeHistory();
     }
   }
 
   public disconnectedCallback() {
     super.disconnectedCallback();
-    this._unsubscribeHistoryTimeWindow();
+    this._unsubscribeHistory();
   }
 
-  private _subscribeHistoryTimeWindow() {
+  private async _subscribeHistory() {
     if (!isComponentLoaded(this.hass!, "history") || this._subscribed) {
       return;
     }
-    this._subscribed = subscribeHistoryStatesTimeWindow(
-      this.hass!,
-      (combinedHistory) => {
-        if (!this._subscribed) {
-          // Message came in before we had a chance to unload
-          return;
-        }
-        this._stateHistory = computeHistory(
-          this.hass!,
-          combinedHistory,
-          this.hass!.localize
-        );
-      },
-      this._hoursToShow,
-      this._entityIds
-    ).catch((err) => {
+    try {
+      this._subscribed = await subscribeHistoryStatesTimeWindow(
```

Comment:
The issue with awaiting is here, is that we will not unsubscribe if the element was disconnected before the promise is resolved. That's why we would store the promise and await it if needed in the unsubscribe function.

Reply:
Ok, that makes sense 👍